inlang/source-code/paraglide/paraglide-js/src/compilerV2/compileBundle.ts

```diff
@@ -0,0 +1,57 @@
+import type { BundleNested, MessageNested } from "@inlang/sdk2"
+import { isValidJSIdentifier } from "../services/valid-js-identifier/index.js"
+import { bundleIndexFunction } from "./bundleIndex.js"
+import { compileMessage } from "./compileMessage.js"
+import { mergeTypeRestrictions, type Compilation } from "./types.js"
+
+type Resource = {
+	/** The compilation result for the bundle index */
+	bundle: Compilation<BundleNested>
+	/** The compilation results for the languages */
+	messages: {
+		[languageTag: string]: Compilation<MessageNested>
+	}
+}
+
+/**
+ * Compiles all the messages in the bundle and returns an index-function + each compiled message
+ */
+export const compileBundle = (
+	bundle: BundleNested,
+	fallbackMap: Record<string, string | undefined>
+): Resource => {
+	if (!isValidJSIdentifier(bundle.id)) {
+		throw new Error(
+			`Cannot compile message with ID "${bundle.id}".\n\nThe message is not a valid JavaScript variable name. Please choose a different ID.\n\nTo detect this issue during linting, use the valid-js-identifier lint rule: https://inlang.com/m/teldgniy/messageLintRule-inlang-validJsIdentifier`
```

Comment:
```suggestion
			`Cannot compile bundle with ID "${bundle.id}".\n\nThe bundle id is not a valid JavaScript variable name. This is likely an internal error. An ID should always be valid. Open an issue in https://github.com/opral/monorepo. 
```

Reply:
The bundle ids are provided by the SDK and should always be valid. This error shouldn't happen./

Reply:
I'll remove the failure case